manifests/server/config.pp

```diff
@@ -98,6 +98,40 @@
         value    => join($::openldap::server::ldapi_ifs, ' '),
         quoted   => 'double',
       }
+
+      if ($openldap::server::provider == 'olc') {
+        # On FreeBSD we need to bootstrap slapd.d
+        $ldif = @(EOL)
+dn: cn=config
+objectClass: olcGlobal
+cn: config
+olcArgsFile: /var/run/openldap/slapd.args
+olcPidFile: /var/run/openldap/slapd.pid
+
+dn: olcDatabase={0}config,cn=config
+objectClass: olcDatabaseConfig
+olcDatabase: {0}config
+olcAccess: to dn.subtree="cn=config" by dn=gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth manage by * none
+
+dn: cn=module,cn=config
+objectClass: olcModuleList
+cn: module
+olcModulepath:  /usr/local/libexec/openldap
+olcModuleload:  back_mdb.la
+olcModuleload:  back_ldap.la
+
+dn: cn=schema,cn=config
+objectClass: olcSchemaConfig
+cn: schema
+
+include: file:///usr/local/etc/openldap/schema/core.ldif
+EOL
+        exec { 'bootstrap cn=config':
+          path    => '/usr/local/sbin',
+          command => "echo '${ldif}' | slapadd -n 0 -F ${openldap::server::confdir}",
+          creates => "${openldap::server::confdir}/cn=config.ldif",
```

Comment:
As much as I dislike `exec`, I don't see an easy way to avoid this at the moment...